conn.go

```diff
@@ -240,7 +252,20 @@ func (c *conn) buildRequest(query string, params []driver.Value, readonly bool)
 		p, _ := c.user.Password()
 		req.SetBasicAuth(c.user.Username(), p)
 	}
-
+	if ctx != nil {
+		quotaKey, quotaOk := ctx.Value(QuotaKey).(string)
+		queryID, queryOk := ctx.Value(QueryID).(string)
+		if quotaOk || queryOk {
+			reqQuery := req.URL.Query()
+			if quotaOk {
```

Comment:
seems like here also can be added check for len(quotaKey) > 0 or whole block can be rewritten like this
```
quotaKey, _ := ctx.Value(QuotaKey).(string)
queryID, _ := ctx.Value(QuotaKey).(string)
if len(quotaKey) > 0 || len(queryID) > 0 {
    reqQuery := req.URL.Query()
    if len(quotaKey) > 0 {
        reqQuery.Add(quotaKeyParamName, quotaKey)
    }
    if len(queryID) > 0 {
        reqQuery.Add(queryIDParamName, queryID)
    }
    req.URL.RawQuery = reqQuery.Encode()
}

Reply:
Thanks for your reply!

As far as I know, Clickhouse supports any strings as quota_key, as well as empty ones. Though, by default it is empty, but if Quota key is specified as additional query param of connection string (which will work for whole connection), there should be a way to set it as empty string during querying (reset to default).
So my idea was to add transparent support - any value for Quota Keys (incl empty ones) and values for Query ID (excl empty ones).

Reply:
ok